ipatests/test_integration/test_ntp_options.py

```diff
@@ -65,131 +55,136 @@ def test_server_client_install_no_ntp(self):
+                                                  extra_args=args)
+            assert expected_msg in server_install.stderr_text
+            cmd = self.master.run_command(['cat', paths.CHRONY_CONF])
+            assert self.ntp_server1 in cmd.stdout_text
+            assert self.ntp_server2 in cmd.stdout_text
+
+            client_install = tasks.install_client(self.master, self.client,
+                                                  extra_args=args)
+            assert expected_msg in client_install.stderr_text
+            cmd = self.client.run_command(['cat', paths.CHRONY_CONF])
+            assert self.ntp_server1 in cmd.stdout_text
+            assert self.ntp_server2 in cmd.stdout_text
 
-        self.cleanup()
+        finally:
+            self.cleanup()
 
     def test_server_replica_client_install_with_pool_and_srv(self):
         """
         test to verify that ipa-server, ipa-replica and ipa-client install
         passes with options --ntp-pool and --ntp-server together
         """
-        ntp_pool = "pool.ntp.org"
-        ntp_server = "1.pool.ntp.org"
         expected_msg = "Configuration of chrony was changed by installer."
+        args = ['--ntp-pool=%s' % self.ntp_pool,
+                '--ntp-server=%s' % self.ntp_server1]
 
-        server_install = tasks.install_master(
-            self.master, setup_dns=False,
-            extra_args=['--ntp-pool=%s' % ntp_pool,
-                        '--ntp-server=%s' % ntp_server])
-        assert expected_msg in server_install.stderr_text
-        cmd = self.master.run_command(['cat', paths.CHRONY_CONF])
-        assert ntp_pool in cmd.stdout_text
-        assert ntp_server in cmd.stdout_text
-
-        replica_install = self.install_replica('--ntp-pool=%s' % ntp_pool,
-                                               '--ntp-server=%s' % ntp_server)
-        assert expected_msg in replica_install.stderr_text
-        cmd = self.replica.run_command(['cat', paths.CHRONY_CONF])
-        assert ntp_pool in cmd.stdout_text
-        assert ntp_server in cmd.stdout_text
-
-        client_install = self.install_client('--ntp-pool=%s' % ntp_pool,
-                                             '--ntp-server=%s' % ntp_server)
-        assert expected_msg in client_install.stderr_text
-        cmd = self.client.run_command(['cat', paths.CHRONY_CONF])
-        assert ntp_pool in cmd.stdout_text
-        assert ntp_server in cmd.stdout_text
-        tasks.uninstall_master(self.replica)
-
-        self.cleanup()
+        try:
+            server_install = tasks.install_master(self.master, setup_dns=False,
+                                                  extra_args=args)
+            assert expected_msg in server_install.stderr_text
+            cmd = self.master.run_command(['cat', paths.CHRONY_CONF])
+            assert self.ntp_pool in cmd.stdout_text
+            assert self.ntp_server1 in cmd.stdout_text
+
+            tasks.install_replica(self.master, self.replica, extra_args=args)
```

Comment:
see previous comment by @varunmylaraiah -- you are replacing one-step replica installation with client + replica installation.

Reply:
And what would you propose ?
should I create new task for installation of replica without promoting client or handle this situation inline in this one tescase?